src/util/logger/Logger.cpp

```diff
@@ -41,7 +40,7 @@ inline std::string NowTime() {
     tm r;
     strftime(buffer, sizeof(buffer), "%X", localtime_r(&tv.tv_sec, &r));
     char result[100];
```

Comment:
```suggestion
    std::array<char,100> result;
```

Reply:
I don't know if that is necessary. `char[100]` is also C++ and the array template has a lot more features than what is needed in `NowTime()`. I wanted to rewrite `NowTime()` entirely in C++ (avoiding POSIX `localtime_r`) but unfortunately C++17 `<chrono>` does not have a way to do this. It seems to be something that C++0a might bring to the table.

Reply:
Both compile to the same binary. The only difference is, that the c++ array enables more static analysis and also provides good debug routines in debug mode. Also some operations are range checked by default.

Reply:
Yes cpp-20 brings a way better time location and calendar support. I would suggest to wait until that day we switch to cpp-20. But this would require to drop some old operating systems like macOS10.13, and Ubuntu 16.04 maybe also 18.04.

Reply:
Okay, to resolve this, I will keep it as `char result[100];`. I'm not feeling like changing it right now because I'm a bit tired from something else.